### PR TITLE
Fix oversize label in no result side menu on iPad

### LINKED ISSUE #19244: Layout problem on empty My Site
### Expected behavior
"Create a new site ..." text should fit the screen.

### Actual behavior
The text doesn't fit the screen. **Only on tablet**

### Steps to reproduce the behavior
1. Launch the app.
2. Log out if you're logged in.
3. Log in with an account without any site or sign up and skip creating the account.
4. Notice the text does not fit on My Site tab.

### PR DESCRIPTION
**Pull request by ipalm0423**
Fixes #19244 #19246
## Description:
This PR fixes the issue where the labels are oversizing in its container view.

• | Before | After
-- | -- | --
iPad | Q1 | Q1-fix
iPad-RTL | Simulator Screen Shot - iPad Pro (9 7-inch) - 2022-09-19 at 01 17 04 | Simulator Screen Shot - iPad Pro (9 7-inch) - 2022-09-19 at 01 12 12

## Root Cause:
In the xib `NoResults.storyboard`, the container view of the labels only has the width constraint `Width <= 360`, but it doesn't have a constraint related to its super view. This will let the labels go outside if its super view is smaller than the label's content.
Constraint on superView
In this case, the side menu view only has `width == 320`. The container view will still go up to `360` because it doesn't care about its super view size. 

## Fix:
- Add missing width constraint between the container view and its super view

## Test Step:
1. Launch the app.
2. Log in with an account without any notification or sign up for a new account.
3. Navigate to the `Notifications` or `My Site` tab.
4. Notice the text should fit inside the left layout.
 
## Regression Notes
1. Potential unintended areas of impact
-  I list all impacted pages in [here](https://docs.google.com/spreadsheets/d/1pjd4YHRTLmDoONsN3gIoTY8pXDOjJuwID0Uipcr1mjg/edit?usp=sharing) with screenshots.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
- Manually test on all impacted pages on iPad, and key pages on iPhone, and list the test result [here](https://docs.google.com/spreadsheets/d/1pjd4YHRTLmDoONsN3gIoTY8pXDOjJuwID0Uipcr1mjg/edit?usp=sharing) 

3. What automated tests I added (or what prevented me from doing so)
add unitTest at 41c4cb3900b9b4893dcabdbace0ca2c7a3d3246d for testing labels width in different sizes of the container.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by twstokes** [comment]
> Any views that present NoResultViewController

👋 @ipalm0423 - do you think we should list out the views that use `NoResultsViewController` so when potentially non-technical testers test these changes, they'll know some to try? Thanks!

**Comment by ipalm0423** [comment]
> > Any views that present NoResultViewController
> 
> 👋 @ipalm0423 - do you think we should list out the views that use `NoResultsViewController` so when potentially non-technical testers test these changes, they'll know some to try? Thanks!

Sure, I created a [view list](https://docs.google.com/spreadsheets/d/1pjd4YHRTLmDoONsN3gIoTY8pXDOjJuwID0Uipcr1mjg/edit?usp=sharing) that is inspected on this PR.

Also, I added some tests in the 41c4cb3900b9b4893dcabdbace0ca2c7a3d3246d. To test the `NoResultsViewController` label width in different width of container.

**Comment by twstokes** [comment]
CI branch: https://github.com/wordpress-mobile/WordPress-iOS/pull/19333

File: WordPress/WordPressTest/ViewRelated/Views/Controllers/NoResultsViewControllerTests.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
+//
+//  NoResultsViewControllerTests.swift
+//  WordPressTest
+//
+//  Created by Jack Chen on 2022/9/18.
+//  Copyright © 2022 WordPress. All rights reserved.
+//
+
```
Comment (twstokes) [inline WordPress/WordPressTest/ViewRelated/Views/Controllers/NoResultsViewControllerTests.swift]: Love these tests!

Would you mind wiping the header here so we match the others?

Comment (ipalm0423) [reply WordPress/WordPressTest/ViewRelated/Views/Controllers/NoResultsViewControllerTests.swift]: Ok, Finished in  33876dc0d336ed3505b78a588fbb420a122b8244

**Review by twstokes — COMMENTED** [review]
This looks really good @ipalm0423! Thanks for being so thorough with your manual tests and screenshots. 🙇 The unit tests are great, too.

I noticed the margins on this view were a little tight. Do you think we should tweak this?

**Comment by ipalm0423** [comment]
> This looks really good @ipalm0423! Thanks for being so thorough with your manual tests and screenshots. 🙇 The unit tests are great, too.
> 
> I noticed the margins on this view were a little tight. Do you think we should tweak this?
> 
> 

Yes, we could tweak the horizontal margin to make it look nice.
Do we have the UI design specification for this margin size? (Figma, Zeplin...etc?)

I only found this [spec](https://dotorgstyleguide.wordpress.com/outline/layout/), but not sure if it's the correct one.

**Comment by twstokes** [comment]
> Yes, we could tweak the horizontal margin to make it look nice. Do we have the UI design specification for this margin size? (Figma, Zeplin...etc?)
> 
> I only found this [spec](https://dotorgstyleguide.wordpress.com/outline/layout/), but not sure if it's the correct one.

Typically we would pull in a designer for Figma designs, but how do you feel about adding just a few points of margin on the sides so it's not as snug?

**Comment by ipalm0423** [comment]
> Typically we would pull in a designer for Figma designs, but how do you feel about adding just a few points of margin on the sides so it's not as snug?

I tweaked the margin for `20` on f05670ec1022b74c5209a3a61933966ddcacd2df , also modify the unit-test

Tested on the narrow key pages:
| margin `20` | default | RTL |
| -- | -- | -- |
| iPhone | iphone13- | Simulator Screen Shot - iPhone 13 - 2022-09-28 at 16 52 54 |
| iPad | iPad9 | ipad9-rtl |

**Comment by twstokes** [comment]
> I tweaked the margin for `20` on [f05670e](https://github.com/wordpress-mobile/WordPress-iOS/commit/f05670ec1022b74c5209a3a61933966ddcacd2df) , also modify the unit-test

Great! Thanks for doing that. I've pushed the latest to the CI branch to run tests. 👍

**Review by twstokes — COMMENTED** [review]
@ipalm0423 can we update `RELEASE-NOTES.txt` for these changes? Thanks!

**Comment by ipalm0423** [comment]
> @ipalm0423 can we update `RELEASE-NOTES.txt` for these changes? Thanks!

No, problem. Rebased to `trunk`, and update the release note at ec96ae642b249bd35cf81b1b8dce3182b8151c3d.

**Comment by twstokes** [comment]
Latest has been pushed to CI branch.

**Review by twstokes — APPROVED** [review]
LGTM @ipalm0423!

**Pull request merged**